runtime/gc_include/ObjectAccessBarrierAPI.hpp

```diff
@@ -436,10 +436,13 @@ class MM_ObjectAccessBarrierAPI
 	VMINLINE void
 	copyObjectFields(J9VMThread *vmThread, J9Class *objectClass, j9object_t srcObject, UDATA srcOffset, j9object_t destObject, UDATA destOffset)
 	{
+		bool hasReferences = J9CLASS_HAS_REFERENCES(objectClass);
 #if defined(J9VM_GC_ALWAYS_CALL_OBJECT_ACCESS_BARRIER)
-		vmThread->javaVM->memoryManagerFunctions->j9gc_objaccess_copyObjectFields(vmThread, objectClass, srcObject, srcOffset, destObject, destOffset);
-#else /* defined(J9VM_GC_ALWAYS_CALL_OBJECT_ACCESS_BARRIER) */
-		if (j9gc_modron_readbar_none != _readBarrierType) {	
+		if (hasReferences) {
+			vmThread->javaVM->memoryManagerFunctions->j9gc_objaccess_copyObjectFields(vmThread, objectClass, srcObject, srcOffset, destObject, destOffset);
+		} else
+#endif /* defined(J9VM_GC_ALWAYS_CALL_OBJECT_ACCESS_BARRIER) */
+		if (hasReferences && (j9gc_modron_readbar_none != _readBarrierType)) {
 			vmThread->javaVM->memoryManagerFunctions->j9gc_objaccess_copyObjectFields(vmThread, objectClass, srcObject, srcOffset, destObject, destOffset);
 		} else {
 			UDATA offset = 0;
```

Comment:
@RSalman This needs to be fixed eventually. This path should not be allowed if WB type is SATB.

Reply:
Noted to address this as part of SATB work